src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect, useCallback, useMemo } from 'react';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setErrorMessage('Something went wrong'))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const setParent = useCallback((arrayOfPeople: Person[]): Person[] => {
+    return arrayOfPeople.map(person => {
+      const mother = arrayOfPeople.find(p => p.name === person.motherName);
+      const father = arrayOfPeople.find(p => p.name === person.fatherName);
+
+      return {
+        ...person,
+        mother,
+        father,
+      };
+    });
+  }, []);
+
+  const preparedPeople = useMemo(() => {
+    return setParent(people);
+  }, [people, setParent]);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="columns is-desktop is-flex-direction-row-reverse">
+          <div className="column is-7-tablet is-narrow-desktop">
+            {!loading && people.length > 0 && <PeopleFilters />}
+          </div>
+
+          <div className="column">
+            <div className="box table-container">
+              {loading && <Loader />}
+
+              {errorMessage && (
+                <p data-cy="peopleLoadingError">{errorMessage}</p>
+              )}
+
+              {!loading && !errorMessage && people.length === 0 && (
+                <p data-cy="noPeopleMessage">
+                  There are no people on the server
+                </p>
+              )}
+
+              {!people.length && !loading && (
+                <p>There are no people matching the current search criteria</p>
+              )}
```

Comment:
The condition `!people.length && !loading` is checked twice. You could simplify this by checking these conditions once and then using nested conditions to check `errorMessage` and `people.length === 0`.